htdocs/sendsms.php

```diff
@@ -96,32 +111,46 @@
     $result = "emptysendsmsform";
 }
 
+# Check the entered username for characters that our installation doesn't support
 if ( $result === "" ) {
-    # Check captcha
-    if ($use_captcha) {
-        $result = global_captcha_check();
-    }
-    # Check the entered username for characters that our installation doesn't support
-    if (  $result === "" ) {
         $result = check_username_validity($login,$login_forbidden_chars);
-    }
+}
+
+#==============================================================================
+# Check captcha
+#==============================================================================
+if ( $result === "" and $use_captcha) {
+        $result = global_captcha_check();
 }
 
 #==============================================================================
 # Check sms
 #==============================================================================
 if ( $result === "" ) {
     [$result, $sms, $displayname] = get_mobile_and_displayname($login);
-
-    if ($sms) {
-        $encrypted_sms_login = encrypt("$sms:$login", $keyphrase);
-        $smsdisplay = $sms;
-        if ( $sms_partially_hide_number ) {
-            $smsdisplay = substr_replace($sms, '****', 4 , 4);
+    if ($sms){
+        if (!$sms_use_ldap) {
+            # TODO: Add possibility to loop over different phone numbers.
+            #foreach ($mailValues as $mailValue) {
+            if (strcasecmp($sms, $phone) == 0) {
+                $match = true;
+                $encrypted_sms_login = encrypt("$sms:$login", $keyphrase);
+                $result = "sendsms";
+            }
+            if (!$match){
+                $result = $obscure_usernotfound_sendtoken ? "tokensent_ifexists" : "smsnomatch";
```

Comment:
If `$obscure_usernotfound_sendtoken` we should always display the screen requesting the SMS code, to avoid indicating to an attacker that the user is not found. We should only display an error if this option is false.

Reply:
This is the new default behaviour, and a new option has been added to the configuration to change into displaying all the information.

Reply:
The new variable is `$obscure_notfound_sendsms = true;`

Reply:
I did some tests, what I see for now:
* the message `smssent_ifexists` is defined in lang files but never used in templates
* the hidden field token is empty if user and/or phone is incorrect, which allows the attacker to know if the search in LDAP fails. We need to generate a token in all cases

Reply:
I've corrected those two problems. The branch have been updated.